gcore/geoheif.cpp

```diff
@@ -86,12 +86,30 @@ void GeoHEIF::setModelTransformation(const uint8_t *payload, size_t length)
 
 CPLErr GeoHEIF::GetGeoTransform(double *padfTransform) const
 {
-    padfTransform[1] = modelTransform[1];
-    padfTransform[2] = modelTransform[2];
-    padfTransform[0] = modelTransform[0];
-    padfTransform[4] = modelTransform[4];
-    padfTransform[5] = modelTransform[5];
-    padfTransform[3] = modelTransform[3];
+    int nAxes = 0;
+    const int *axes = has_SRS()
+                          ? OSRGetDataAxisToSRSAxisMapping(
+                                OGRSpatialReference::ToHandle(&m_oSRS), &nAxes)
+                          : nullptr;
+
+    if (axes && axes[0] != 1)
```

Comment:
@rouault , @desruisseux mentioned:

> The axes[0] != 1 check should be abs(axes[0]) != 1, because GDAL use negative numbers as a convention meaning that axis direction should be reversed (e.g. south -> north).

is this the case that `GetDataAxisToSRSAxisMapping()` could return negative values here?

Thanks!

Reply:
> is this the case that `GetDataAxisToSRSAxisMapping()` could return negative values here?

in theory yes, but I'm not smart enough to have figured out what the OAMS_TRADITIONAL_GIS_ORDER should be for the weird southing,westing or westing,southing CRS in the context of raster data should be (at least in the GeoTIFF context, people use the CRS variants that use northing-easting or easting-northing axis order to avoid those headaches. southing,westing or westing,southing are left only for purely geodesy non-mapping use cases), so in practice you won't get negative values, unless you manually set them with SetDataAxisToSRSAxisMapping(). I would probably not worry too much about that...